### ISSUE TITLE
False positive for Style/ArgumentsForwarding when only a block argument can be forwarded anonymously

### ISSUE THREAD

**Issue by mvz** [rubocop/rubocop#12618]
In the new RuboCop version 1.60.0, the Style/ArgumentsForwarding cop now autocorrects argument lists with only a single block argument, or positional arguments combined with a block argument, to use `...` instead of the block argument. While this does not change things when the method is called in the original way, it does change the arity and will now forward (additional) positional and keyword arguments.

## Expected behavior

Given the following methods:
```ruby
def foo(&block)
  bar(&block)
end

def baz(qux, quuz, &block)
  bar(qux, quuz, &block)
end
```

I expect RuboCop _not_ to autocorrect when targeting Ruby before 3.1.

When targeting 3.1+, I expect it to autocorrect to:

```ruby
def foo(&)
  bar(&)
end

def baz(qux, quuz, &)
  bar(qux, quuz, &)
end
```

## Actual behavior

RuboCop autocorrects as follows, regardless of whether Ruby 3.0 or 3.1 is targeted.

```ruby
def foo(...)
  bar(...)
end

def baz(qux, quuz, ...)
  bar(qux, quuz, ...)
end
```

## Steps to reproduce the problem

1. Create a file `foo.rb` with the content given in the first code block above.
2. Run `rubocop -A`

## RuboCop version

```
$ bundle exec rubocop -V
1.60.0 (using Parser 3.3.0.3, rubocop-ast 1.30.0, running on ruby 3.3.0) [x86_64-linux]
  - rubocop-performance 1.20.2
  - rubocop-rspec 2.26.1
```

**Comment by koic** [comment]
I cannot reproduce the issue. Can you provide an executable repro step?

```console
$ cat example.rb
def foo(&block)
  bar(&block)
end

def baz(qux, quuz, &block)
  bar(qux, quuz, &block)
end
```

```yaml
$ cat .rubocop.yml
AllCops:
  TargetRubyVersion: 3.2
```

```console
$ bundle exec rubocop --only Style/ArgumentsForwarding
(snip)

Inspecting 1 file
.

1 file inspected, no offenses detected
```

**Comment by mvz** [comment]
Hi @koic, thanks for looking into this. Here's an executable reproduction:
```ruby
$ cat example.rb
def foo(&block)
  bar(&block)
end

def baz(qux, quuz, &block)
  bar(qux, quuz, &block)
end
```

```yaml
$ cat .rubocop.yml
AllCops:
  TargetRubyVersion: 3.1
```

```console
$ rubocop example.rb --only Style/ArgumentsForwarding
[...]
Inspecting 1 file
C

Offenses:

example.rb:1:9: C: [Correctable] Style/ArgumentsForwarding: Use shorthand syntax ... for arguments forwarding.
def foo(&block)
        ^^^^^^
example.rb:2:7: C: [Correctable] Style/ArgumentsForwarding: Use shorthand syntax ... for arguments forwarding.
  bar(&block)
      ^^^^^^
example.rb:5:20: C: [Correctable] Style/ArgumentsForwarding: Use shorthand syntax ... for arguments forwarding.
def baz(qux, quuz, &block)
                   ^^^^^^
example.rb:6:18: C: [Correctable] Style/ArgumentsForwarding: Use shorthand syntax ... for arguments forwarding.
  bar(qux, quuz, &block)
                 ^^^^^^
```

**Comment by mvz** [comment]
I think the changes that result in this behavior were introduced in #12541, where some RuboCop code was also (wrongly, I think) autocorrected.

**Comment by dvandersluis** [comment]
@mvz I see what you're saying but I think it is actually the case prior to my change as well - certain autocorrects will change the arity of the method.

For example (this example converts `*args, &` to `...` which means kwargs are now accepted):

https://github.com/rubocop/rubocop/blob/f0c62ef587e8cef80ddf5345648d15ad92768902/spec/rubocop/cop/style/arguments_forwarding_spec.rb#L759-L773

Similarly - converting just `**kwargs` to `...` means that args are now accepted:

https://github.com/rubocop/rubocop/blob/f0c62ef587e8cef80ddf5345648d15ad92768902/spec/rubocop/cop/style/arguments_forwarding_spec.rb#L397-L411

Am I correct that these are the same issue?

**Comment by mvz** [comment]
> Am I correct that these are the same issue?

I think the first one is the same or else very similar. The second one is behind a non-default setting (`AllowOnlyRestArgument: false`), so that would be a conscious choice to allow a potential change in behavior.

**Comment by mvz** [comment]
Thanks, @koic!

**Comment by stanley90** [comment]
Not sure it's okay, just tried with 1.60.1 and

```ruby
  def parse_file(file, parse_method = determine_parse_method(file.path), &)
    parse_method, args = Array.wrap(parse_method)
    send(parse_method, file, args, &)
  end
```

gives

```
...: C: [Correctable] Style/ArgumentsForwarding: Use anonymous block arguments forwarding (&).
  def parse_file(file, parse_method = determine_parse_method(file.path), &)
                                                                         ^
...: C: [Correctable] Style/ArgumentsForwarding: Use anonymous block arguments forwarding (&).
    send(parse_method, file, args, &)
                                   ^
```

**Comment by koic** [comment]
@stanley90 Oops! This is a bug. I've opened #12626 to solve the false positive.

**Comment by mvz** [comment]
Hmm .. I think I'm still seeing this with RuboCop 1.60.1: https://github.com/troessner/reek/actions/runs/7559358324/job/20582995886?pr=1764.

**Comment by koic** [comment]
@mvz Thank you for the feedback. I've opened #12628 to solve the false positive.
